elephant/cubic.py

```diff
@@ -0,0 +1,212 @@
+'''
+CuBIC is a statistical method for the detection of higher order of
+correlation in of parallel spike train based on the analysis of the
+cumulants of the population count.
+Given a list sts of SpikeTrains, the analysis comprises the following
+steps:
+1) compute the population histogram (PSTH) with the desired bin size
+   >>> binsize = 5 * pq.ms
+   >>> pop_count = elephant.statistics.time_histogram(sts, binsize)
+2) apply CuBIC to the population count
+   >>> aplha = 0.05  # significance level of the tests used
+   >>> xi, p_val, k = cubic(data, ximax=100, alpha=0.05,\
+           errorval=4.):
+'''
+# -*- coding: utf-8 -*-
+from __future__ import division
+import scipy.stats
+import scipy.special
+import math
+import warnings
+
+
+# Based on matlab code by Benjamin Staude
+# Adaptation to python by Pietro Quaglio and Emiliano Torre
+
+
+def cubic(data, ximax=100, alpha=0.05):
+    '''
+    Performs the CuBIC analysis [1] on a population histogram, calculated from
+    a population of spiking neurons.
+
+    The null hypothesis :math:`H_0: k_3(data)<=k^*_{3,\\xi}` is iteratively
+    tested with increasing correlation order :math:`\\xi` (correspondent to
+    variable xi) until it is possible accept, with a significance level alpha,
+    that :math:`\\hat{\\xi}` (correspondent to variable xi_hat) is the minimum
+    order of correlation necessary to explain the third cumulant
+    :math:`k_3(data)`.
+
+    :math:`k^*_{3,\\xi}` is the maximized third cumulant, supposed a CPP model
```

Comment:
CPP stands for Compound Poisson Process, it is a model for correlated spike trains proposed in Staude et al. 2009